core/store/src/test/java/uk/gov/gchq/gaffer/store/operation/handler/output/ToOpenCypherCsvHandlerTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.store.operation.handler.output;
+
+import com.google.common.collect.Lists;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.gov.gchq.gaffer.commonutil.TestGroups;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.output.ToOpenCypherCsv;
+import uk.gov.gchq.gaffer.serialisation.implementation.StringSerialiser;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.SchemaEdgeDefinition;
+import uk.gov.gchq.gaffer.store.schema.SchemaEntityDefinition;
+import uk.gov.gchq.gaffer.store.schema.TypeDefinition;
+import uk.gov.gchq.koryphe.impl.binaryoperator.StringConcat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ToOpenCypherCsvHandlerTest {
+
+    @Mock
+    Store storeMock;
+
+    @Test
+    public void shouldConvertToOpenCypherCsvWithHeader() throws OperationException {
+        // Given
+        final List<Element> elements = Lists.newArrayList(
+                makeEntity("vertex1", "count", 1),
+                makeEntity("vertex2"),
+                makeEdge("source1", "count", 1),
+                makeEdge("source2")
+        );
+
+        final ToOpenCypherCsv operation = new ToOpenCypherCsv.Builder()
+                .input(elements)
+                .neo4jFormat(false)
+                .build();
+
+        final ToOpenCypherCsvHandler handler = new ToOpenCypherCsvHandler();
+
+        //When
+        when(storeMock.getSchema()).thenReturn(makeSchema());
+        final Iterable<? extends String> results = handler.doOperation(operation, new Context(), storeMock);
+
+        //Then
+        final List<String> resultList = Lists.newArrayList(results);
+        final List<String> expected = Arrays.asList(
+                ":ID,:LABEL,:TYPE,:START_ID,:END_ID,count:Integer,DIRECTED:Boolean",
```

Comment:
Currently this says `Integer` rather than `Int`, but it will be fixed by https://github.com/gchq/Gaffer/issues/2726